### ISSUE TITLE
feat: enable tslib helpers

### ISSUE THREAD

**Issue by stephenlautier** [dherges/ng-packagr#337]
## Type of Issue

```
[ ] Bug Report
[x] Feature Request
```

## Description
Not sure if you already have considered this, however, doesn't make sense to enable tslib helpers by default? if not some kind of config override might be needed in order to allow this.

all it needs

1.
```json
    "compilerOptions": {
        "importHelpers": true
    }
```
2. add `tslib` as dependency

https://github.com/Microsoft/tslib

#### Expected Behaviour

tslib helpers not emitted in the output, as this will make the build smaller.

#### Version Information

```
ng-packagr: v2.0.0-rc.4
```

**Comment by dherges** [comment]
Hi @stephenlautier,

considered, yes. Aren't the `__ decorate`, `__extends`, and so on helpers not required to be included at least once in the esm5 bundle?

You made me check Angular's GitHub builds repos and `@angular/common` is:

FESM'5 Bundle:

`
``js
import { __assign, __extends } from 'tslib';
```

https://github.com/angular/common-builds/blob/master/esm5/common.js

UMD Bundle:

```js
var extendStatics = Object.setPrototypeOf ||
    ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
    function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };

function __extends(d, b) {
    extendStatics(d, b);
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
}

var __assign = Object.assign || function __assign(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
    }
    return t;
};
```

https://github.com/angular/common-builds/blob/master/bundles/common.umd.js#L32-L44

---

TL;DR: you're right ... the quickest way to get there is to set `"importHelpers": true` and then let rollup inline the tslib helpers to the top of the bundles.

**Comment by dherges** [comment]
Can you review #337 and take a look at the Circle CI artefacts?

**Comment by stephenlautier** [comment]
Sure I will check it out 👍 

Not sure if it make sense to add `tslib` as a `dependency` in the generated `package.json`, or you recommend simply have to it to add it to your main `package.json` manually?

**Comment by stephenlautier** [comment]
@dherges artifacts for integration samples are not uploaded, not sure if its because the build is failing tho

```
Uploading /tmp/samples/integration/samples to samples
  No artifact files found at /tmp/samples/integration/samples
```

https://circleci.com/gh/dherges/ng-packagr/682#artifacts/containers/0

**Comment by alan-agius4** [comment]
I think in general ‘tslib’ iis an dependency, at least the last time i checked angular where do it as such

**Comment by dherges** [comment]
@stephenlautier Tests are green again. Can you take a look?

@alan-agius4 I am not sure. From the `@angular/core` builds: in the UMD bundle `tslib` is inlined. In ESM5 it's imported. In ESM2015, it's not even present because their usage subset of TypeScript is convergent to EcmaScript 2015.

- UMD: https://github.com/angular/core-builds/blob/master/bundles/core.umd.js#L12-L44
- ESM5: https://github.com/angular/core-builds/blob/master/esm5/core.js#L6
- ESM2015: https://github.com/angular/core-builds/blob/master/esm2015/core.js

**Comment by alan-agius4** [comment]
@dherges if `importHelpers` will be done by default, in the `package.json` when amending it, it should also include.
```ts
  "dependencies": {
    "tslib": "^1.7.1"
  },
```
https://github.com/angular/core-builds/blob/master/package.json#L12

**Comment by dherges** [comment]
And then inline it only in the UMD?

https://github.com/angular/material2-builds/blob/master/bundles/material-button.umd.js

**Comment by alan-agius4** [comment]
Yes
